src/scrabble.js

```diff
@@ -1,5 +1,98 @@
-function scrabble() {
+const scoreValues = {
+  // Corresponding alphabet to letter score values
+  A: 1,
+  E: 1,
+  I: 1,
+  O: 1,
+  U: 1,
+  L: 1,
+  N: 1,
+  R: 1,
+  S: 1,
+  T: 1,
+  D: 2,
+  G: 2,
+  B: 3,
+  C: 3,
+  M: 3,
+  P: 3,
+  F: 4,
+  H: 4,
+  V: 4,
+  W: 4,
+  Y: 4,
+  K: 5,
+  J: 8,
+  X: 8,
+  Q: 10,
+  Z: 10,
+}
 
+function isValidInput(word) {
+  return (word !== undefined && word !== null);
 }
 
-module.exports = scrabble
+function scrabble(word) {
+  if (!isValidInput(word)) {
+    return 0;
+  }
+
+  let s = 0; // stands for score 
+  let wordMultiplier = 1;
+  let i = 0;
+
+  while (i < word.length) {
+    let letter = word[i].toUpperCase();
+
+    if (letter === '{' || letter === '[') {
+      let closingChar = letter === '{' ? '}' : ']';
+      let j = word.indexOf(closingChar, i + 1);
+      if (j === -1) {
+        return 0;
+      }
+      let letters = word.substring(i + 1, j).toUpperCase();
+      let letterMultiplier = letter === '{' ? 2 : 3;
+      for (let k = 0; k < letters.length; k++) {
+        let letterScore = scoreValues[letters[k].toUpperCase()] || 0;
+        s += letterScore * letterMultiplier;
+        console.log(`letter: ${letters[k]}, multiplier: ${letterMultiplier}, score: ${letterScore}`);
+      }
+      i = j + 1;
+    } else if (letter === '|') {
+      let j = word.indexOf('|', i + 1);
+      if (j === -1 || (j - i) % 2 === 0) {
+        return 0;
+      }
+      let letters = word.substring(i + 1, j).toUpperCase();
+      let letterMultiplier = 1;
+      for (let k = 0; k < letters.length; k++) {
+        let letterScore = scoreValues[letters[k].toUpperCase()] || 0;
```

Comment:
If you're getting to the point of using k as a variable then you're probably going a bit too deep with the nested loops.